lib/lints/avoid_returning_widgets/avoid_returning_widgets_rule.dart

```diff
@@ -20,22 +20,23 @@ class AvoidReturningWidgetsRule extends DartLintRule {
   ) {
     context.registry.addDeclaration((node) {
       final bool isWidgetReturned;
-      if (node is FunctionDeclaration) {
-        final returnType = node.returnType?.type;
-        isWidgetReturned = returnType != null && hasWidgetType(returnType);
-      } else if (node is MethodDeclaration) {
-        // `build` methods return widgets by nature
-        if (node.declaredElement?.name == "build") {
-          return;
-        }
+      switch (node) {
+        case (final FunctionDeclaration node):
+          final returnType = node.returnType?.type;
+          isWidgetReturned = returnType != null && hasWidgetType(returnType);
+        case (final MethodDeclaration node):
+          // `build` methods return widgets by nature
+          if (node.declaredElement?.name == "build") {
+            return;
+          }
 
-        // `FunctionDeclaration` and `MethodDeclaration` implement
-        // different things to have this `returnType` getter so
-        // unfortunately it can't be simplified.
-        final returnType = node.returnType?.type;
-        isWidgetReturned = returnType != null && hasWidgetType(returnType);
-      } else {
-        isWidgetReturned = false;
+          // `FunctionDeclaration` and `MethodDeclaration` implement
+          // different things to have this `returnType` getter so
+          // unfortunately it can't be simplified.
+          final returnType = node.returnType?.type;
+          isWidgetReturned = returnType != null && hasWidgetType(returnType);
+        default:
+          isWidgetReturned = false;
```

Comment:
I was thinking something like this. Cuts down a few null checks but looks... not quite readable.

@n-bernat what do you think.
```suggestion
      switch (node) {
        case (FunctionDeclaration(
            returnType: TypeAnnotation(:final type?),
          )):
          isWidgetReturned = hasWidgetType(type);
        case (MethodDeclaration(
            returnType: TypeAnnotation(:final type?),
            declaredElement: ExecutableElement(:final name)
          )):
          // `build` methods return widgets by nature
          if (name == "build") return;

          // `FunctionDeclaration` and `MethodDeclaration` implement
          // different things to have this `returnType` getter so
          // unfortunately it can't be simplified.
          isWidgetReturned = hasWidgetType(type);
        default:
          isWidgetReturned = false;
```

Reply:
That's actually quite good. A lot less clutter, nice and to the point. Let's do this.